fastlane_bot/tools/invariants/functions/__init__.py

```diff
@@ -0,0 +1,35 @@
+"""
+Represents a function ``y = f(x; params)`` and vectors thereof
+
+This module contains two classes, ``Function`` and ``FunctionVector``. 
+
+-   The ``Function`` class represents a function of the form ``y = f(x; params)``,
+    where ``x`` is the input value and ``params`` are arbitrary additional
+    parameters fed into the (data)class upon instantiation.
+
+-   The ``FunctionVector`` class represents a vector (linear combination) of
+    ``Function`` objects, and implements a function interface (via pointwise
+    evaluation), a vector interface (from the ``DictVector`` inheritance). A
+    ``FunctionVector`` also contains an integration kernel, which allows it to
+    expose a number of norms and distance measures.
+    
+TODO: other imported objects eg ``DerivativeFunction``, ``Derivative2Function``
+
+---
+(c) Copyright Bprotocol foundation 2024. 
+Licensed under MIT
+"""
+from .core import __VERSION__, __DATE__
+
+# objects defined in core
+from .core import Function, FunctionVector
+from .core import PriceFunction, Price2Function
+from .core import minimize, goalseek
+from .core import fmt
+
+# objects defined in funcs and funcsAMM
+from .funcs import *
+from .funcsAMM import *
```

Comment:
I'm always in favor of importing exactly what's needed (for readability, not other impact as far as I'm aware of).

Reply:
I generally agree but those files are designed to _only_ contain example functions, and I simply want to import all of them without having to adjust the import statement whenever I add examples